query/src/org/labkey/query/controllers/QueryController.java

```diff
@@ -4548,12 +4548,16 @@ protected JSONObject executeJson(JSONObject json, CommandType commandType, boole
             {
                 if (behaviorType != null && behaviorType != AuditBehaviorType.NONE)
                 {
-                    if (transaction.getAuditEvent() != null)
-                        auditEvent = transaction.getAuditEvent();
+                    DbScope.Transaction auditTransaction = transacted ? transaction : table.getSchema().getScope().getCurrentTransaction();
+                    if (auditTransaction == null)
+                        auditTransaction = transaction;

```

Comment:
I'm not sure I understand this. If `transacted` is true, we use the ensured transaction from the try block and if it's false we get the transaction from the table schema. Presumably if it's transacted, these are the same thing. If there is no transaction on the table schema, we use the `NO_OP_TRANSACTION`.  So could this be
`
```suggestion
DbScope.Transaction auditTransaction = table.getSchema().getScope().getCurrentTransaction();
if (auditTransaction == null)
    auditTransaction = NO_OP_TRANSACTION;
```

Reply:
I've modified the logic a little bit more to only get scope transaction when the outer caller is indeed transacted.